test/jasmine/tests/parcoords_test.js

```diff
@@ -45,6 +45,23 @@ function purgeGraphDiv(done) {
     return delay(50)().then(done);
 }
 
+function getAvgPixelByChannel(id) {
+    var canvas = d3.select(id).node();
+
+    var imgData = readPixel(canvas, 0, 0, canvas.width, canvas.height);
+    var n = imgData.length * 0.25;
```

Comment:
A minor change here:
instead of dividing by 4, we use multiply by 0.25 so that we return rgb fractions and avoid possibility of rounding small values to 0.

Reply:
shouldn't cause problems but... is there actually a difference between these in js? anyway `n` should always be an integer, right?

https://github.com/plotly/plotly.js/blob/458f5938a538517922afb08790a164339e8d1970/test/jasmine/assets/read_pixel.js#L11

Reply:
I'm curious. Can you share an example of this happening in JS?

Reply:
(am I missing something? why didn't that link get turned into a code snippet?)

Reply:
You are totally right and I tested this on IE, FF and Chrome too. 
In JS there is actually no difference in JS between multiplying by 0.25.
Thanks for the note.

Reply:
> (am I missing something? why didn't that link get turned into a code snippet?)

![image](https://user-images.githubusercontent.com/6675409/47854904-6bad2e00-ddb9-11e8-96a4-4cb61cd813d0.png)

it's rendering fine now. I'll blame the hiccup on Microsoft. 

> Thanks for the note.

Thank you for double checking!